internal/core/command/init.go

```diff
@@ -15,218 +15,103 @@
-				LoggingClient = logger.NewClient(clients.CoreCommandServiceKey, Configuration.Logging.EnableRemote, logTarget, Configuration.Writable.LogLevel)
-
-				//Initialize service clients
-				initializeClients(useRegistry)
-			}
-		}
 
-		//Only attempt to connect to database if configuration has been populated
-		if Configuration != nil {
-			err := connectToDatabase()
-			if err != nil {
-				ch <- err
-			} else {
-				break
-			}
-		}
-		time.Sleep(time.Second * time.Duration(1))
-	}
-	close(ch)
-	wait.Done()
-
-	return
-}
-
-func Init(useRegistry bool) bool {
-	if Configuration == nil {
-		return false
-	}
-
-	if useRegistry {
-		registryErrors = make(chan error)
-		registryUpdates = make(chan interface{})
-		go listenForConfigChanges()
-	}
-
-	go telemetry.StartCpuUsageAverage()
-
-	return true
+type server interface {
+	IsRunning() bool
 }
 
-func Destruct() {
-	if registryErrors != nil {
-		close(registryErrors)
-	}
-
-	if registryUpdates != nil {
-		close(registryUpdates)
-	}
+// ServiceInit encapsulates information needed for the service to startup.
+type ServiceInit struct {
+	server server
 }
 
-func initializeConfiguration(useRegistry bool, configDir, profileDir string) (*ConfigurationStruct, error) {
-	//We currently have to load configuration from filesystem first in order to obtain RegistryHost/Port
-	configuration := &ConfigurationStruct{}
-	err := config.LoadFromFile(configDir, profileDir, configuration)
-	if err != nil {
-		return nil, err
-	}
-	configuration.Registry = config.OverrideFromEnvironment(configuration.Registry)
+// BootstrapHandler fulfills the BootstrapHandler contract and performs and startup actions needed by the service.
+func (s ServiceInit) BootstrapHandler(
+	wg *sync.WaitGroup,
+	ctx context.Context,
+	startupTimer startup.Timer,
+	config bootstrap.Configuration,
+	logging logger.LoggingClient,
+	registry registry.Client) bool {
 
-	if useRegistry {
-		err = connectToRegistry(configuration)
-		if err != nil {
-			return nil, err
-		}
+	// update global variables.
+	LoggingClient = logging
 
-		rawConfig, err := registryClient.GetConfiguration(configuration)
-		if err != nil {
-			return nil, fmt.Errorf("could not get configuration from Registry: %v", err.Error())
-		}
+	// initialize clients required by service.
+	s.initializeClients(registry != nil, registry)
 
-		actual, ok := rawConfig.(*ConfigurationStruct)
-		if !ok {
-			return nil, fmt.Errorf("configuration from Registry failed type check")
-		}
-
-		configuration = actual
-
-		// Check that information was successfully read from Registry
-		if configuration.Service.Port == 0 {
-			return nil, errors.New("error reading configuration from Registry")
+	// initialize database.
+	for startupTimer.HasNotElapsed() {
+		var err error
+		dbClient, err = s.newDBClient(Configuration.Databases["Primary"].Type)
+		if err == nil {
+			break
 		}
+		LoggingClient.Warn(fmt.Sprintf("couldn't create database client: %v", err.Error()))
```

Comment:
You need a `dbClient = nil` statement before the logging statement.  See https://github.com/edgexfoundry/edgex-go/issues/1827